integrade/tests/api/v1/test_cloud_accounts.py

```diff
@@ -0,0 +1,279 @@
+"""Tests for cloud accounts.
+
+:caseautomation: automated
+:casecomponent: api
+:caseimportance: high
+:caselevel: integration
+:requirement: Cloud Meter
+:testtype: functional
+:upstream: yes
+"""
+from urllib.parse import urljoin
+
+import pytest
+
+from integrade import api
+from integrade import config
+from integrade.tests.api.v1 import urls
+from integrade.tests.api.v1.utils import (
+    create_user_account,
+    get_auth,
+)
+from integrade.utils import gen_password, uuid4
+from integrade.utils import needs_valid_cloud_accounts
+
+
+@pytest.mark.serial_only
+@needs_valid_cloud_accounts
+def test_create_cloud_account(drop_account_data):
+    """Ensure cloud accounts can be registered to a user.
+
+    :id: d0174576-9b7c-48f7-8556-b560badf062d
+    :description: Ensure an user can register a cloud account by specifying
+        the role ARN.
+    :steps: 1) Create a user and authenticate with their password
+        2) Send a POST with the cloud account information to 'api/v1/account/'
+        3) Send a GET to 'api/v1/account/' to get a list of the cloud accounts
+        4) Attempt to create a duplicate and expect it to be rejected
+        5) Attempt to delete the account and expect to be rejected
+    :expectedresults: The server returns a 201 response with the information of
+        the created account, and the account cannot be duplicated or deleted.
+    """
+    user = create_user_account({
+        'email': '',
+        'password': gen_password(),
+        'username': uuid4(),
+    })
+    auth = get_auth(user)
+    client = api.Client(authenticate=False)
+    # make sure we are authenticated
+    client.get(urls.AUTH_ME, auth=auth)
+
+    cfg = config.get_config()
+    cloud_account = {
+        'account_arn': cfg['valid_role'],
+        'resourcetype': 'AwsAccount'
+    }
+    create_response = client.post(
+        urls.API_CLOUD_ACCOUNT,
+        payload=cloud_account,
+        auth=auth
+    )
+    assert create_response.status_code == 201
+
+    acct = create_response.json()
+
+    # get specific account
+    get_response = client.get(
+        urljoin(urls.API_CLOUD_ACCOUNT,
+                '{}/'.format(acct['id'])
+                ), auth=auth)
+    assert acct == get_response.json()
+
+    # list cloud accounts associated with this user
+    list_response = client.get(urls.API_CLOUD_ACCOUNT, auth=auth)
+    assert acct in list_response.json()['results']
+
+    # TODO need to try and update name, but
+    # feature is not delivered yet.
+
+    # TODO need to try and update arn, but
+    # feature is not delivered yet.
+
+    # assert we cannot create duplicate
+    client.response_handler = api.echo_handler
+    response = client.post(
+        urls.API_CLOUD_ACCOUNT,
+        payload=cloud_account,
+        auth=auth
+    )
+    assert response.status_code == 400
```

Comment:
Does it return a message stating it is duplicated? If yes, do you think asserting for it would bring any value?